docs/getting_started_jax.md

```diff
@@ -0,0 +1,252 @@
+
+1. What has changed in JAX
+2. Why Shardy?
+3. Future plans
+
+This is meant for JAX users who use `jax.jit` for running training/inference models across more than 1 GPU or TPU (batch parallelism, megatron, ZeRO, etc). They would be using things like `PartitionSpec`s and `NamedSharding`s.
+
+## 1. What has changed in JAX?
+
+### State of JAX before: GSPMD
+
+Prior to Shardy, JAX users who partitioned their models across models across multiple devices used [GSPMD](http://go/arxiv/2105.04663) behind the scenes.
+
+GSPMD is the propagation+partitioning system that lives in the middle of the XLA pipeline. It operates on HLO - the IR that comes after StableHLO (the program you get after running `jax.jit.lower`).
+
+JAX doesn't run GSPMD directly, but encodes instructions into the StableHLO IR.
+
+But before we go any further, let's introduce our working example.
+
+Note that this example should be executed in a TPU environment with 8 devices.
+
+```python
+import os
+# make sure our program runs on 8 devices
+os.environ['XLA_FLAGS'] = '--xla_force_host_platform_device_count=8'
+
+import jax
+import numpy as np
+from jax import numpy as jnp
+from jax.sharding import Mesh, NamedSharding, PartitionSpec
+from jax._src.sharding_impls import UNSPECIFIED
+from jax.experimental.shard_map import shard_map
+```
+
+First let's create our mesh.
+
+```python
+mesh = Mesh(
+    np.reshape(np.array(jax.devices()), (4, 2)),
+    ('data', 'model'))
+
+print(mesh.shape)
+```
+
+### In/Out shardings
+
+Let's look what changed the most: how sharding attributes are encoded in the JAX program for the compiler to read.
+
+Let's look at it through an example. It's going to be an MLP-like model consisting of no bias tensors, and 2 layers (two matmuls).
+
+```python
+def predict(x, w1, w2):
+  x = jnp.tanh(x)
+  z1 = jnp.einsum('ij,jk->ik', x, w1)
+  z2 = jnp.einsum('ij,jk->ik', z1, w2)
+  return jnp.sin(z2)
+```
+
+What we will want to do here sharding wise is:
+
+1. `data` parallelism on x
+2. `model` parallelism on w1 and w2 through the [megatron](http://go/arxiv/1909.08053) sharding strategy.
+
+Now let's prepare the model for GSPMD sharding. Note that we will explicitly shard `w1`, but let GSPMD propagation shard `w2`.
+
+```python
+def run_in_out_shardings():
+  samples = jax.ShapeDtypeStruct((16, 128), jnp.float32, sharding=NamedSharding(mesh, PartitionSpec('data', None)))
+  samples_sharding = NamedSharding(mesh, PartitionSpec('data', None))
+  w1 = jax.ShapeDtypeStruct((128, 256), jnp.float32, sharding=NamedSharding(mesh, PartitionSpec(None, 'model')))
+  w1_sharding = NamedSharding(mesh, PartitionSpec(None, 'model'))
+  w2 = jax.ShapeDtypeStruct((256, 10), jnp.float32)
+  w2_sharding = UNSPECIFIED
+
+  print(jax.jit(predict, in_shardings=(samples_sharding, w1_sharding, w2_sharding)).lower(samples, w1, w2).as_text())
+
+run_in_out_shardings()
+```
+
+GSPMD's sharding annotations are defined as [HloShardingV2](TBA). So we have the following correspondance:
```

Comment:
Should we add content on the HloShardingV2 format in a separate page and link here?

Reply:
I will replace with GSPMD sharding here, too - following what we did on the Colab.